### PR TITLE
Feature alerts

### LINKED ISSUE #25: Use ngx-bootstrap for popovers

### PR DESCRIPTION
**Pull request by codeepic**
Resolve #25 - use ngx-bootstrap popover for `feature alert` functionality.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     "@angular/cdk": "^5.2.5",
     "@angular/common": "^5.0.0",
     "@angular/core": "^5.0.0",
-    "@angular/forms": "^5.0.0"
+    "@angular/forms": "^5.0.0",
+    "ngx-bootxtrap": "2.0.4"
```
Comment (garethdn) [inline package.json]: Might be worth prepending this with `^`?

File: src/_lib/modules/feature-alerts/feature-alerts.module.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { NgModule, ModuleWithProviders } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FeatureAlertComponent } from './feature-alert.component';
+import { FeatureAlertsService } from './feature-alerts.service';
+import { WindowRef } from './windowref';
+import {PopoverModule} from 'ngx-bootstrap';
```
Comment (garethdn) [inline src/_lib/modules/feature-alerts/feature-alerts.module.ts]: Just for uniformity, space after opening bracket and before closing bracket.

Also import from `ngx-bootstrap/popover`

File: src/_lib/sass/src/_popovers.scss [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // Popovers
 // --------------------------------------------------
 
+$popover-border-color: get-color('gray', 5x-light);
```
Comment (garethdn) [inline src/_lib/sass/src/_popovers.scss]: Should we not just update the `$popover-border-color` in variables.scss?

File: src/_lib/sass/src/_popovers.scss [hunk 4]
```diff
@@ -14,119 +15,166 @@
...
     background-clip: padding-box;
-    // border: 1px solid $popover-fallback-border-color;
-    // border: 1px solid $popover-border-color;
-    // border-radius: $border-radius-large;
-    // @include box-shadow(0 5px 10px rgba(0,0,0,.2));
-  
+    border: 1px solid $popover-border-color;
+    border-radius: 3px;
+    box-shadow: 0 0 8px rgba(0,0,0,.2);
+    font-family: 'Proxima Nova Regular', Arial, sans-serif;
```
Comment (garethdn) [inline src/_lib/sass/src/_popovers.scss]: I don't think this line is necessary if Analytics has something like the following set:

```
$font-family-sans-serif:  "Helvetica Neue", Helvetica, Arial, sans-serif !default;
$font-family-base:        $font-family-sans-serif !default;
```

File: src/_lib/sass/src/_popovers.scss [hunk 5]
```diff
@@ -14,119 +15,166 @@
...
+    z-index: 999;
```
Comment (garethdn) [inline src/_lib/sass/src/_popovers.scss]: Is there a reason for z-index 999? We have a master list of z-index set in the variables file, with popovers currently at `1060`. This is for controlling how certain elements appear in relation to each, e.g popovers within modals, toasts over modals etc

File: src/_lib/sass/styles.scss [hunk 6]
```diff
@@ -83,6 +83,7 @@
 @import "src/navs";
 @import "src/scaffolding";
 @import "src/popovers";
+@import "src/feature-alerts";
```
Comment (garethdn) [inline src/_lib/sass/styles.scss]: I see this as more of an addition than a replacement

File: src/_lib/sass/src/_popovers.scss [hunk 7]
```diff
@@ -14,119 +15,166 @@
...
-      left: 50%;
-      margin-left: -$popover-arrow-outer-width;
-      border-bottom-width: 0;
-      border-top-color: $popover-arrow-outer-fallback-color; // IE8 fallback
-      border-top-color: $popover-arrow-outer-color;
-      bottom: -$popover-arrow-outer-width;
-      &:after {
+}
+
+.popover.top > .arrow {
```
Comment (garethdn) [inline src/_lib/sass/src/_popovers.scss]: Is there any reason why these styles have changed from 

```
.popover {
    &.top > .arrow {
```

to

`.popover.top > .arrow {`

File: src/app/app.module.ts [hunk 8]
```diff
@@ -61,6 +65,9 @@ import { PickerComponent } from './picker/picker.component';
     NwPickerModule,
     ToastsModule.forRoot(),
     MarkdownModule.forRoot(),
+    //PopoverModule.forRoot(), //ngx-bootstrap - no longer needed
```
Comment (garethdn) [inline src/app/app.module.ts]: Remove?

File: package.json (hunk 1 above)
Comment (codeepic) [reply package.json]: Done.

File: src/_lib/modules/feature-alerts/feature-alerts.module.ts (hunk 2 above)
Comment (codeepic) [reply src/_lib/modules/feature-alerts/feature-alerts.module.ts]: Done.

File: src/_lib/sass/src/_popovers.scss (hunk 3 above)
Comment (codeepic) [reply src/_lib/sass/src/_popovers.scss]: Yep.

File: src/_lib/sass/src/_popovers.scss (hunk 4 above)
Comment (codeepic) [reply src/_lib/sass/src/_popovers.scss]: Removed.

File: src/_lib/sass/src/_popovers.scss (hunk 5 above)
Comment (codeepic) [reply src/_lib/sass/src/_popovers.scss]: We had some issues in analytics with popover appearing above the dash dropdown.

Comment (codeepic) [reply src/_lib/sass/src/_popovers.scss]: Removed - I can deal with it in analytics.

File: src/_lib/sass/styles.scss (hunk 6 above)
Comment (codeepic) [reply src/_lib/sass/styles.scss]: It is a small addition that holds the width of the btn.

File: src/_lib/sass/src/_popovers.scss (hunk 7 above)
Comment (codeepic) [reply src/_lib/sass/src/_popovers.scss]: Just went through the styles and thought that nesting a lot of styles only to prepend the `.popover` in front of them when sass is compiled is of little value. Not nesting in that case offers improved legibility.

File: src/app/app.module.ts (hunk 8 above)
Comment (codeepic) [reply src/app/app.module.ts]: Yep.

File: src/_lib/sass/styles.scss (hunk 6 above)
Comment (codeepic) [reply src/_lib/sass/styles.scss]: ok, I see what you mean, I moved it down.

**Pull request merged**